include/boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp

```diff
@@ -100,23 +100,40 @@ class range_segment_iterator
             Reference
         >
 {
+private:
+    static bool has_less_than_two_elements(Range const& r)
+    {
+        iterator_type first = range_iterator_begin<Range>::apply(r);
+        iterator_type last = range_iterator_end<Range>::apply(r);
+
+        return first == last || ++first == last;
+    }
+
 public:
     typedef typename range_iterator_type<Range>::type iterator_type;
 
     // default constructor
     range_segment_iterator()
-        : m_it()
+        : m_it(), m_has_less_than_two_elements()
```

Comment:
As far as I know, you choose the T() initialization when you want to say something like "if it's an object it's supposed to know how to properly initialize itself (and I'm not supposed to know) and if it's a fundamental type then I know I want it to be 0 (which the C++ standard guarantees)". Iterators are a typical example: if it's a raw pointer you want 0, if it's an object you want it to do whatever is needed for it to be ready for iteration. So you write T() (you'll see it everywhere in the STL).

In the case you are manipulating something that's supposed to represent a scalar value, and you want it to "mathematically represent" X, then you are not really in the situation described above. You're saying "whatever you are, you are now equal to X". So you'll rather use T(X).

This is all general guidelines of course.

Bruno